transmat/mixins/fshash/bucket_memory.go

```diff
@@ -13,15 +13,46 @@ var _ Bucket = &MemoryBucket{}
 
 type MemoryBucket struct {
 	// my kingdom for a red-black tree or other sane sorted map implementation
-	lines []Record
+	names   []string
+	records map[string]Record
 }
 
 func (b *MemoryBucket) AddRecord(metadata fs.Metadata, contentHash []byte) {
+	if b.records == nil {
+		b.records = map[string]Record{}
+	}
+
+	name := metadata.Name.String()
+	if metadata.Type == fs.Type_Dir {
+		name += "/"
+	}
```

Comment:
no action requested -- but looking over how often this comes up: I have to say, if I ever wrote packages like this again, I'd most certainly have the metadata struct hold dir name strings *with* the trailing slash... because it'd be a lot easier to substring that off the end when you actively don't want it, than to post-append it every time you do want it.